src/core/symbology/qgscategorizedsymbolrenderer.h

```diff
@@ -379,11 +379,14 @@ class CORE_EXPORT QgsCategorizedSymbolRenderer : public QgsFeatureRenderer
     QString legendClassificationAttribute() const override { return classAttribute(); }
 
     /**
-     * creates a QgsCategorizedSymbolRenderer from an existing renderer.
+     * Creates a new QgsCategorizedSymbolRenderer from an existing \a renderer.
+     *
+     * The optional \a layer parameter is required for conversions of some renderer types.
```

Comment:
Should we add a note on when this was added, e.g. "Since QGIS 3.20, the optional \a layer..."?